nucypher-core/src/reencryption.rs

```diff
@@ -41,39 +42,61 @@ impl ReencryptionRequest {
         encrypted_kfrag: &EncryptedKeyFrag,
         publisher_verifying_key: &PublicKey,
         bob_verifying_key: &PublicKey,
-        conditions: Option<&[u8]>,
-        context: Option<&[u8]>,
+        conditions: Option<&Conditions>,
+        context: Option<&Context>,
     ) -> Self {
         Self {
             capsules: capsules.into(),
             hrac: *hrac,
             encrypted_kfrag: encrypted_kfrag.clone(),
             publisher_verifying_key: *publisher_verifying_key,
             bob_verifying_key: *bob_verifying_key,
-            conditions: conditions.map(|c| c.into()),
-            context: context.map(|c| c.into()),
+            conditions: conditions.cloned(),
+            context: context.cloned(),
         }
     }
 }
 
+#[derive(Deserialize)]
+struct ReencryptionRequest0 {
```

Comment:
What's the `0` for?

Reply:
Short for `ReencryptionRequest_v1_0` - defines the old format we can deserialize from. `clippy` complains about `_v1_0`, so I shortened it, seeing as how we can only support different minor versions, so we won't need to disambiguate between `v1` and `v2`.